sway/input/seatop_down.c

```diff
@@ -110,3 +188,22 @@ void seatop_begin_down(struct sway_seat *seat, struct sway_container *con,
 	container_raise_floating(con);
 	transaction_commit_dirty();
 }
+
+void seatop_begin_touch_down(struct sway_seat *seat, struct sway_container *con,
+		struct wlr_event_touch_down *event, double sx, double sy) {
+	struct sway_cursor *cursor = seat->cursor;
+
+	struct seatop_down_event *e = calloc(1, sizeof(struct seatop_down_event));
+	sway_assert(e, "Unable to allocate seatop_down_event");
```

Comment:
We can reduce this duplication by:

```c
seatop_begin_down(seat, con, sx, sy);
handle_touch_down(seat, event);
```

You can either get rid of the `time_msec` parameter to `seatop_down` (it's not needed anywhere), or just pass 0 to it for now.

That'll also raise floating containers on touch, which we want.